misc/services/mobile-webui/mobile-webui-frontend/src/reducers/wfProcesses_status/picking.js

```diff
@@ -148,12 +256,100 @@ const normalizePickingLines = (lines) => {
       ...line,
       steps: line.steps.reduce((accum, step) => {
         accum[step.pickingStepId] = step;
+        accum[step.pickingStepId].altSteps = { qtyToPick: 0, genSteps: {} };
+
+        // Mock generated steps - used for testing w/o real data
+        // accum[step.pickingStepId].altSteps.genSteps = {
+        //   1000819: {
+        //     id: '1000819',
+        //     locatorName: 'Hauptlager',
+        //     huBarcode: '1000437',
+        //     uom: 'Kg',
+        //     qtyAvailable: 45,
+        //     qtyPicked: 0,
+        //   },
+        //   1000820: {
+        //     id: '1000820',
+        //     locatorName: 'Hauptlager',
+        //     huBarcode: '1000463',
+        //     uom: 'Kg',
+        //     qtyAvailable: 25,
+        //     qtyPicked: 0,
+        //   },
+        // };
+
         return accum;
       }, {}),
     };
   });
 };
 
+// TODO: use single generating fnct for alt steps - (DRY) - when no time pressure, do the gen only on activity
+const generateAlternativeStepsPostBackendFetch = ({ draftDataStored, lineId, stepId, qtyToAllocate }) => {
+  console.log('draftActivityDataStored in generate:', original(draftDataStored));
+  const draftDataStoredOrig = original(draftDataStored);
+  const stepData = draftDataStoredOrig.lines[lineId].steps[stepId];
+  const draftStep = draftDataStored.lines[lineId].steps[stepId];
+  const { pickFromAlternatives: alternativesPool } = draftDataStoredOrig;
+
+  console.log('qtyToAllocate ===>', qtyToAllocate);
+
+  let qtyToAllocateRemaining = qtyToAllocate;
+
+  // loop among the items and for the ones that do have `qtyPicked` -> for each we allocate to the pool (pickFromAlternatives in the root of dataStored)
+  for (let [keyAltStep, altStepItem] of Object.entries(stepData.pickFromAlternatives)) {
```

Comment:
where are you copying the qtyPicked got from backend to genSteps ?

Pls do the following test:
* pick something
* clear your application data
* refresh from backend
If it's nicely restored => OK
If not => NOK


IMHO when you sync from backend you have to:
* clear genSteps
* Consider what backend said that it was picked and create the genSteps with qtys picked provided by backend
* run **the function** which allocates which:
    * allocates the qtyPicked against the pool
    * check how much is still needed to be picked and generates more genSteps if needed

That **the function** shall be the function which u have to run after user is picking/unpicking something.


This is how I am seeing the things,
but pls don't take it for granted because might have errors in my thoughts.
Pls grasp it.